pulumi/src/index.ts

```diff
@@ -58,11 +66,34 @@ export interface ServiceConfig {
   storageSize: string
 }
 
+export interface ServiceInput {
+  coinServiceName: string
+  ports: Record<string, Port>
+  command?: Array<string>
+  args?: Array<string>
+  env?: Record<string, string>
+  dataDir?: string
+  configMapData?: Record<string, string>
+  volumeMounts?: Array<k8s.types.input.core.v1.VolumeMount>
+  readinessProbe?: k8s.types.input.core.v1.Probe
+  livenessProbe?: k8s.types.input.core.v1.Probe
+}
+
+export interface JointCoinServiceInput extends ServiceInput, ServiceConfig {}
+
 export interface Service {
+  name: string
   ports: Array<
     k8s.types.input.core.v1.ServicePort & { ingressRoute?: boolean; pathPrefix?: string; stripPathPrefix?: boolean }
   >
   configMapData: Record<string, string>
   containers: Array<k8s.types.input.core.v1.Container>
   volumeClaimTemplates: Array<k8s.types.input.core.v1.PersistentVolumeClaim>
 }
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type Outputs = Record<string, any>
+
+export type SecretData = pulumi.Input<{
```

Comment:
Don't really need this typing. The secret data func can just return `Record<string, string>` (pulumi allows regular types to be used for Input<> vars)